README.adoc

```diff
@@ -213,17 +209,17 @@ Setting this to *true* enables all of the following:
 * `strictPropertyInitialization`: Makes sure you properly initialize class properties _(helpful when extending Foundry's document classes)_
 * `noImplicitThis`: Prevents confusion about what `this` refers to _(particularly important in Foundry's hook callbacks)_
 
-noUncheckedIndexedAccess:: Requires you to verify that array elements or object properties exist before using them.
+noUncheckedIndexedAccess:: Adds `undefined` to types with index signatures. Index signatures are like catch-alls for particular shapes of keys. For example an array in TypeScript will allow expressions like `arr[10]` because that key may exist at runtime. However with `noUncheckedIndexedAccess` set to false (the default) the resulting types will not include `undefined` even though at runtime such a key may not exist.
 +
 Setting this to *true* forces you to check whether a value accessed from an array or object exists (i.e. is not `undefined`) before you attempt to use it. This prevents runtime errors when accessing dynamic Foundry data like `actor.items[0]` or `token.data.flags["my-module"]`, which might not always exist.
 
-noUncheckedSideEffectImports:: Ensures TypeScript resolves all import paths, even if they don't import anything that requires type-checking. By default, TypeScript will ignore imports that don't resolve to any values (i.e. files imported for their top-level "side effects", like registering hooks or defining properties in the global scope).
+noUncheckedSideEffectImports:: Ensures TypeScript does not silently allow invalid import paths when no values or types are imported from it. For example `import "./foo/bar";` would not error if that path does not really exist but `import { foo } from "./foo/bar";` would.
```

Comment:
I think this one is a bit too confusing as well; how about this:

> noUncheckedSideEffectImports:: Ensures TypeScript validates the path of every import statement. By default, TypeScript validates import paths only when explicit values or types are being imported (e.g. `import {foo} from "./foo/bar"`), silently allowing import paths that do not (e.g. `import "./foo/bar"`) without comfirming they point to a valid file.
> +
> Setting this to *true* ensures all imports must resolve correctly, improving code reliability.

Reply:
You good with:
> Ensures TypeScript validates the path of every import statement. By default, TypeScript validates import paths only when explicit values or types are being imported, e.g. `import {foo} from "./foo/bar"`. This means that imports like `import "./foo/bar" may not exist.
>
> Setting this to true ensures all imports must resolve correctly, improving code reliability.

Some of this is just flow about the parenthetical `e.g.`s but I also did some things like changing the language from "may not point to a valid file" because not all imports point to 'files' because stuff like `declare module` exists and breaks that assumption. I intentionally left the language vague about what an import _does_ point to because I don't intend to dive into how import resolution works.